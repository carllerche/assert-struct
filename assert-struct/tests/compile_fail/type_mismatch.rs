use assert_struct::assert_struct;

#[derive(Debug)]
struct User {
    name: String,
    age: u32,
}

fn main() {
    let user = User {
        name: "Alice".to_string(),
        age: 30,
    };

    // This should NOT compile - age is u32, not String
    assert_struct!(user, User {
        name: "Alice",
        age: "thirty",
    });
}