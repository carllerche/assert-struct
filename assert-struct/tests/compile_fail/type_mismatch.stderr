error[E0277]: can't compare `u32` with `String`
  --> tests/compile_fail/type_mismatch.rs:18:14
   |
18 |         age: "thirty",
   |              ^^^^^^^^ no implementation for `u32 == String`
   |
   = help: the trait `PartialEq<String>` is not implemented for `u32`
           but trait `PartialEq<u32>` is implemented for it
   = help: for that trait implementation, expected `u32`, found `String`
   = note: required for `&u32` to implement `PartialEq<&String>`
